fmralign/metrics.py

```diff
@@ -0,0 +1,180 @@
+import numpy as np
+from scipy.stats import pearsonr
+from sklearn.metrics import r2_score
+
+
+def score_table(ground_truth, prediction, masker, loss,
+                multioutput='raw_values'):
+    """
+    Calculates loss function for predicted, ground truth
+    arrays. Supported scores are R2, correlation, and normalized
+    reconstruction error (Bazeille et al., 2019)
+
+    Parameters
+    ----------
+    ground_truth: 3D or 4D Niimg
+        Reference image
+    prediction : 3D or 4D Niimg
+        Same shape as `ground_truth`
+    masker: instance of NiftiMasker or MultiNiftiMasker
+        Masker to be used on ground_truth and prediction. For more information see:
+        http://nilearn.github.io/manipulating_images/masker_objects.html
+    loss : str in ['R2', 'corr', 'n_reconstruction_err']
+        The loss function used in scoring. Default is normalized
+        reconstruction error.
+        'R2' :
+            The R2 distance between source and target arrays.
+            Best possible score is 1.0 and it can be negative (because the
+            model can be arbitrarily worse).
+        'corr' :
+            The correlation between source and target arrays.
+        'n_reconstruction_err' :
+            The normalized reconstruction error. A perfect prediction
+            yields a value of 1.0
+    multioutput: str in ['raw_values', 'uniform_average']
+        Defines aggregating of multiple output scores. Default is raw values.
```

Comment:
applies only if loss is not 'corr'

Reply:
no apply for all metrics since they are all voxelwise (with time dimension as samples) and the averaging here is across voxels)

Reply:
But if I'm not mistaken this is not used in the code (when loss == 'corr'), is it ?

Reply:
Yes you're right, only the multioutput case is currently handled

Reply:
I think that the docstring should be a bit mroe rpecise here.

Reply:
it was added in the code since then so now everything is consistent and docstring is right I think